llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -6807,8 +6807,11 @@ static Expected<Function *> createOutlinedFunction(
     OMPBuilder.ConstantAllocaRaiseCandidates.emplace_back(Func);
 
   // Insert target deinit call in the device compilation pass.
-  llvm::OpenMPIRBuilder::InsertPointOrErrorTy AfterIP =
-      CBFunc(Builder.saveIP(), Builder.saveIP());
+  BasicBlock *OutlinedBodyBB =
+      splitBB(Builder, /*CreateBranch=*/true, "outlined.body");
+  llvm::OpenMPIRBuilder::InsertPointOrErrorTy AfterIP = CBFunc(
+      Builder.saveIP(),
+      OpenMPIRBuilder::InsertPointTy(OutlinedBodyBB, OutlinedBodyBB->begin()));
```

Comment:
Does clang/can clang use OMPIRBuilder for `target`? This will change their generated code. Maybe somebody who contributes to clang should review.

Reply:
So far, it is only referenced from the MLIR translation layer. However, I added @ddpagan from the clang side.

Reply:
Thanks!